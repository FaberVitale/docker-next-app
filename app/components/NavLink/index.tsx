import React, { FC, HTMLAttributes } from 'react';
import Link, { LinkProps as NextLinkProps } from 'next/link';

export type NavProps = NextLinkProps & HTMLAttributes<HTMLAnchorElement>;

const NavLink: FC<NavProps> = ({
  href,
  as,
  replace,
  scroll,
  shallow,
  passHref,
  prefetch,
  ...rest
}) => (
  <Link
    href={href}
    as={as}
    replace={replace}
    shallow={shallow}
    scroll={scroll}
    passHref={passHref}
    prefetch={prefetch}
  >
    {/* eslint-disable jsx-a11y/anchor-has-content */}
    <a {...rest} />
    {/* eslint-enable jsx-a11y/anchor-has-content */}
  </Link>
);

export default NavLink;
