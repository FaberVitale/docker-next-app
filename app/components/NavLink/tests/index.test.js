import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import NavLink from '..';

afterEach(cleanup);

describe('NavLink', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <NavLink
        href="/about.asp"
        onClick={() => {}}
        rel="noopener"
        title="about-title"
      >
        anchor-text
      </NavLink>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
