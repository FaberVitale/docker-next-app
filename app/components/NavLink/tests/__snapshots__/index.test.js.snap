// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink matches snapshot 1`] = `
<DocumentFragment>
  <a
    href="/about.asp"
    rel="noopener"
    title="about-title"
  >
    anchor-text
  </a>
</DocumentFragment>
`;
